### PR TITLE
CV2-5247-support-yake-keyword-extraction-for-chinese

### PR DESCRIPTION
**Pull request by ahmednasserswe**
## Description
using jieba to segment mandarin text

Reference: CV2-5247 and CV2-4909
## How has this been tested?
Has it been tested locally? Are there automated tests?
locally. will add automated tests as well. 

## Are there any external dependencies?
Are there changes required in sysops terraform for this feature or fix?

## Have you considered secure coding practices when writing this code?
Please list any security concerns that may be relevant.

### CODE REVIEW

**Review by skyemeedan — CHANGES_REQUESTED** [review]
Can you add a test with a text example that shows this working?  Ideally would be a chunk of non space delimited Chinese text that is badly parsed without jieba and then returns appropriate keywords when processed with jieba.

**Review by skyemeedan — APPROVED** [review]
seems like a really cool library!

File: lib/model/yake_keywords.py [hunk 1]
```diff
@@ -36,6 +37,8 @@ def normalize_special_characters(self, text):
             text = text.replace(k, v)
         return text
 
+    def run_chinese_segmentation_with_jieba(self, text):
+        return " ".join(list(jieba.cut_for_search(text)))
```
Comment (DGaffney) [inline lib/model/yake_keywords.py:41]: add newline after function please

**Comment by DGaffney** [comment]
One extremely minor note otherwise good to go!

**Review by DGaffney — APPROVED** [review]

File: lib/model/yake_keywords.py [hunk 2]
```diff
@@ -58,6 +61,10 @@ def run_yake(self, text: str,
             language = cld3.get_language(text).language
         ### normalize special characters
         text = self.normalize_special_characters(text)
+        # Segmentation for mandarin
+        if language == 'zh-CN' or language == 'zh' or language == 'zh-TW':
```
Comment (computermacgyver) [inline lib/model/yake_keywords.py:65]: Let's change to `if language[:2]=="zh"`

File: lib/model/yake_keywords.py (hunk 1 above)
Comment (computermacgyver) [reply lib/model/yake_keywords.py:41]: I don't see a need for this to be a separate function given it's one line and only invoked in one place. I'd just include `text = " ".join(list(jieba.cut_for_search(text)))` right in the `run_yake` function

File: lib/model/yake_keywords.py [hunk 3]
```diff
@@ -58,6 +61,10 @@ def run_yake(self, text: str,
...
+            text = self.run_chinese_segmentation_with_jieba(text)
+            # text = " ".join(list(jieba.cut_for_search(text)))
```
Comment (computermacgyver) [inline lib/model/yake_keywords.py:67]: Per above, I would drop the function call and just use `text = " ".join(list(jieba.cut_for_search(text)))` here. If we do stick with the function call, then please remove the commented code.

File: lib/model/yake_keywords.py (hunk 1 above)
Comment (ahmednasserswe) [reply lib/model/yake_keywords.py:41]: I wanted to put it in a separate function so I can write tests separately for it. namely `test_run_chinese_segmentation_with_jieba` As this step includes downloading and loading a model. Is that the clean way to do so?

File: lib/model/yake_keywords.py (hunk 3 above)
Comment (ahmednasserswe) [reply lib/model/yake_keywords.py:68]: added my comment about this above. please, Let me know what do you think about it.

Comment (computermacgyver) [reply lib/model/yake_keywords.py:68]: Sounds good. Let's leave as is.

**Pull request merged**